honeybadger_extensions/flask.py

```diff
@@ -66,31 +66,41 @@ def _patch_generic_request_payload(self):
         """
         def generic_request_payload_decorator(original):
             def _wrapper(request, context, config):
-                current_view = current_app.view_functions[_request.endpoint]
-                if hasattr(current_view, 'view_class'):
-                    component = '.'.join((current_view.__module__, current_view.view_class.__name__))
-                else:
-                    component = current_view.__module__
-                payload = {
-                    'url': _request.base_url,
-                    'component': component,
-                    'action': _request.endpoint,
-                    'params': {},
-                    'session': filter_dict(dict(session), config.params_filters),
-                    'cgi_data': {
-                        k: v
-                        for k, v in iteritems(_request.headers)
-                        if k not in self.skip_headers
-                    },
-                    'context': context
-                }
-
-                # Add query params
-                params = filter_dict(dict(_request.args), config.params_filters)
-                params.update(filter_dict(dict(_request.form), config.params_filters))
-                payload['params'] = params
-
-                return payload
+                try:
+                    endpoint = _request.endpoint
+                    view_functions = current_app.view_functions
+                    current_view = view_functions[endpoint]
+                    if hasattr(current_view, 'view_class'):
+                        component = '.'.join((current_view.__module__, current_view.view_class.__name__))
+                    else:
+                        component = current_view.__module__
+
+                    payload = {
+                        'url': _request.base_url,
+                        'component': component,
+                        'action': _request.endpoint,
+                        'params': {},
+                        'session': filter_dict(dict(session), config.params_filters),
+                        'cgi_data': {
+                            k: v
+                            for k, v in iteritems(_request.headers)
+                            if k not in self.skip_headers
+                        },
+                        'context': context
+                    }
+
+                    # Add query params
+                    params = filter_dict(dict(_request.args), config.params_filters)
+                    params.update(filter_dict(dict(_request.form), config.params_filters))
+                    payload['params'] = params
+
+                    return payload
+                except Exception:
+                    logger.exception('Error while generating generic request payload')
```

Comment:
This is expected every time we alert sth by bg threads so it is going to be too noisy.

Let's log it, instead, as a debug log.